src/scss/_main.scss

```diff
@@ -352,31 +373,37 @@
 	@include oGridColspan($span);
 }
 
+@mixin oGridRowMargins($total-cols: $o-grid-columns) {
+	$subgrid-ratio: $o-grid-columns / $total-cols;
+
+	margin-right: auto;
+	margin-left: auto;
+}
+
 /// Base row styles
 ///
 /// @param {String} $grid-mode [$o-grid-mode]
-@mixin oGridRow($grid-mode: $o-grid-mode) {
+@mixin oGridRow($grid-mode: $o-grid-mode, $total-cols: $o-grid-columns) {
 	clear: both;
 	min-width: $o-grid-min-width;
 	// Older browsers get a fixed-width layout
 	max-width: oGridGetMaxWidthForLayout($o-grid-fixed-layout);
-	padding-left: $o-grid-gutter / 2;
-	padding-right: $o-grid-gutter / 2;
 	box-sizing: border-box;
-	margin-left: auto;
-	margin-right: auto;
+
+	@include oGridRowMargins($total-cols);
 
 	// Clearfix
 	zoom: 1;
+	overflow: auto;
```

Comment:
Children need to be able to bleed out.

One use case is the dropdown menus such as the ones in the header (see http://build.origami.ft.com/files/o-header@3.0.6/demos/header.html).

We need both `:before` and `:after` to avoid margin collapsing of children that aren't floated of "flex". It provides a consistent implementation across modern browsers and IE8/9.